assets/js/admin/payment-method-restrictions.js

```diff
@@ -0,0 +1,96 @@
+jQuery( function( $ ) {
+
+	/**
+	 * If the WC core validation passes (errors removed), check our own validation.
+	 */
+	$( document.body ).on( 'wc_remove_error_tip', function( e, element, removed_error_type ) {
+		var type_error = 'i18n_zero_subscription_error';
+
+		// Exit early if it's the zero subscription error that has been removed.
+		if ( removed_error_type === type_error ) {
+			return;
+		}
+
+		var product_type = $( '#product-type' ).val();
+
+		if ( 'subscription' !== product_type && 'variable-subscription' !== product_type ) {
+			return;
+		}
+
+		// We're only interested in the product's recurring price and sale price input.
+		if ( 'subscription' === product_type && ! $( element ).is( '#_subscription_price' ) && ! $( element ).is( '#_sale_price' ) ) {
+			return;
+		}
+
+		if ( 'variable-subscription' === product_type && ! $( element ).hasClass( 'wc_input_subscription_price' ) && ! $( element ).is( '.wc_input_price[name^=variable_sale_price]' ) ) {
+			return;
+		}
+
+		// Reformat the product price - remove the decimal place separator and remove excess decimal places.
+		var price = accounting.unformat( $( element ).val(), wcs_gateway_restrictions.decimal_point_separator );
+		price     = accounting.formatNumber( price, wcs_gateway_restrictions.number_of_decimal_places );
+
+		if ( 0 >= price ) {
+			$( document.body ).triggerHandler( 'wc_subscriptions_add_error_tip', [ element, type_error ] );
+		} else {
+			$( document.body ).triggerHandler( 'wc_remove_error_tip', [ element, type_error ] );
+		}
+	} );
+
+	/**
+	 * Validate the recurring price or sale price field on element change event or when a validate event is triggered.
+	 */
+	$( document.body ).on( 'change wc_subscriptions_validate_zero_recurring_price', '#_subscription_price, #_sale_price, .wc_input_subscription_price, .wc_input_price[name^=variable_sale_price]', function() {
```

Comment:
The variable product sale price elements don't have a unique targetable class. Here I've used `.wc_input_price[name^=variable_sale_price]` which is the same as WC core does to [target variable product sale elements](https://github.com/woocommerce/woocommerce/blob/5.9.0/assets/js/admin/woocommerce_admin.js#L137).